[WIP] Heatmap example

jmkiley opened the pull request:
Ported [this example](https://www.mapbox.com/mapbox-gl-js/example/heatmap/) from GL-JS. I switched the data source to use the [USGS earthquake live data](https://earthquake.usgs.gov/earthquakes/feed/v1.0/geojson.php).

- [X] Write example
- [x] Add comments
- [ ] Incorporate revisions

<img width="313" alt="screenshot 2017-05-12 11 30 13" src="https://cloud.githubusercontent.com/assets/12474734/26012837/bdb8c030-370a-11e7-8bb0-18d90ed4ac36.png">

friedbunny commented on Examples/Swift/PointHeatmapExample.swift:
```diff
@@ -0,0 +1,54 @@
+        
+        // Parse GeoJSON data from USGS on earthquakes in the past week.
+        DispatchQueue.global().async {
+            guard let url = URL(string: "https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_week.geojson") else { return }
```
This URL should never fail to initialize and can be safely force-unwrapped.

friedbunny replied:
The URLs themselves are also different — the objc example uses the static GL-JS version of this feed.

friedbunny commented on Examples/Swift/PointHeatmapExample.swift:
```diff
@@ -0,0 +1,54 @@
+            guard let url = URL(string: "https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_week.geojson") else { return }
+            DispatchQueue.main.async {
+                self.displayEarthquakes(url: url, style: style)
+            }
```
Can you say more about what the intention is here, re: threading? It appears that you’re creating the URL on a background thread, then doing the actual fetching and rendering on the main thread.

friedbunny commented on Examples/Swift/PointHeatmapExample.swift:
```diff
@@ -0,0 +1,54 @@
+                       MGLShapeSourceOption.clusterRadius: 20,
+                       MGLShapeSourceOption.maximumZoomLevel: 15] as [MGLShapeSourceOption : Any]
+        
+        let earthquakeSource = MGLShapeSource(identifier: "earthquakes", url: url, options: options)
```
If this is where the remote fetching happens, should there be some protection against it failing?

jmkiley replied:
I changed the source to a Mapbox hosted one. Will look into using this live data source for a tutorial/different example.

jmkiley commented:
<img width="310" alt="screenshot 2017-05-23 10 20 09" src="https://cloud.githubusercontent.com/assets/12474734/26416393/911646b0-406a-11e7-8676-7b7afcd872e8.png">

Changed the opacity and colors, switched to dark style.

jmkiley commented:
Is there anything else to address with this? @friedbunny @captainbarbosa

captainbarbosa commented:
No, I don't think so @jmkiley - looks good! 👍
